### DISCUSSION TITLE
Is there a way to receive click events from charts?

### DISCUSSION THREAD

**Discussion by jzeimen** [Avaiga/taipy#299] [Q&A] [answered]
Let's say I have a pie chart summarizing a table below it of tasks. The pie chart is showing the percentage of done vs not done items. 

I want to be able to click on the "not done" section of the pie chart and then filter the table to only include "not done" items.

**Comment by FlorianJacta** [comment, marked as answer]
You have this option for the majority of charts. The property is called `selected` but is unavailable for pie charts. I created a quick example to show you how to use it in a bar chart. 

Click on the bar until you see it is selected; the table should be refreshed accordingly.

```python
from taipy.gui import Gui, State
import pandas as pd

data = pd.DataFrame({"values":[60, 40],
                     "labels":["Done", "Not Done"]})

data_for_table = data
selected = []

md = """
<|{data}|chart|type=bar|y=values|x=labels|selected={selected}|>

<|{data_for_table}|table|>
"""

def on_change(state: State, var_name: str, var_value):
    if var_name == "selected":
        if len(var_value) == 1:
            label = state.data.loc[var_value[0], 'labels']
            state.data_for_table = state.data.where(state.data["labels"] == label)
        else:
            state.data_for_table = state.data

Gui(page=md).run()
```

![image](https://github.com/Avaiga/taipy/assets/98709993/d6cb3c8b-360f-4f89-8e57-bd0967c3ca79)


For your use case with a pie chart, I would suggest using a [selector](https://docs.taipy.io/en/latest/manuals/gui/viselements/selector/) or [toggle](https://docs.taipy.io/en/latest/manuals/gui/viselements/toggle/) instead. We will investigate expanding the selected property to pie charts.

**Reply by jzeimen** [reply]
Ah ok thanks, I could use a bar chart instead

**Reply by FlorianJacta** [reply]
Yes, you could. We are currently investigating whether we can expand it to pie charts for your use case. ([Issue](https://github.com/Avaiga/taipy-gui/issues/919))
